### PR TITLE
ei_iter safety blanket

### PR DESCRIPTION
**Pull request by aridf**
This contains a couple small changes to `ei_iter`, most notably:

- Setting `ei::ei` to run three times trying different values of `erho` before stopping the function. Users are notified about the occurrence of errors and the next value of `erho` being tried. 
- Turning off the `samples` parameter that allows users to choose how many importance samples they want to take during the sampling portion of `ei::ei`. It was causing results to deviate from `ei_est_gen` by as much as 10%. I'm not sure why this was happening, but it's off for now. 

Still to add:

- [x]  Finalized measures of uncertainty on plots.

### CODE REVIEW

**Comment by lorenc5** [comment]
This makes sense to me. To be clear the 3 run on erho is to run it each time or only if it busts keep running with switching erho numbers low and high to see if it will execute?

File: R/ei_iter.R [hunk 1]
```diff
@@ -126,6 +133,8 @@ ei_iter <- function(
     }
   }
 
+  if (verbose) message(paste("Beginning", n_iters, "2x2 estimations..."))
```
Comment (pssachdeva) [inline R/ei_iter.R:136]: This should be on two lines, with braces.

File: R/ei_iter.R [hunk 2]
```diff
@@ -152,24 +161,103 @@ ei_iter <- function(
...
-            formula = formula,
-            total = totals_col,
-            erho = erho,
-            simulate = FALSE,
-            # args_pass
-          )
+    # Calls to ei are wrapped in nested try-catches that try different erho
+    # parameters to get the function working. The underlying ei function is
+    # unstable and changing erho can break it.
+    tryCatch(
```
Comment (pssachdeva) [inline R/ei_iter.R:167]: I wonder if there's a way to do this without nested try catch statements. Ultimately it makes the code harder to read and is harder to work on top of. 

One alternative is to use a while/for loop, iterating over possible `erho` values. Then, exit the loop once it passes without error.

File: R/ei_iter.R [hunk 3]
```diff
@@ -152,24 +161,103 @@ ei_iter <- function(
...
+            tryCatch(
+              {
+                utils::capture.output({
+                  ei_out <-
+                    suppressMessages(
+                      purrr::lift(ei::ei)(
+                        data = data,
+                        formula = formula,
+                        total = totals_col,
+                        erho = 20,
```
Comment (pssachdeva) [inline R/ei_iter.R:232]: I think this should be 0.5.

**Comment by pssachdeva** [comment]
I think there are commits in here from previous pull requests, which is why the commit history is so long. This might be because it branched off from a previous branch. It won't matter if you squash and merge, but you can also periodically [rebase](https://git-scm.com/docs/git-rebase) onto master in order to keep the commit history clean.

**Comment by aridf** [comment]
> Is removing the `samples` parameter necessary? If it was returning different values from `ei_est_gen` even using the default value, then it makes sense. But if it gives different results for different numbers of samples, it can remain as a default value. An issue can be made to look into it further.

@lorenc5 and I found cases where the functions deviated by as much as 10% using the same sample size, and I was able to make the deviation go to a negligible size with this change. I will create an issue to look into it further.

**Comment by aridf** [comment]
> This makes sense to me. To be clear the 3 run on erho is to run it each time or only if it busts keep running with switching erho numbers low and high to see if it will execute?

If the function works as expected, `ei::ei` is only run once. If it returns an error, `ei::ei` is called again with `erho = 0.5`. If it fails again, it is called one more time with `erho = 20`. After that, if it still fails, it just returns an error.

File: R/ei_iter.R (hunk 3 above)
Comment (aridf) [reply R/ei_iter.R]: fixed

**Comment by pssachdeva** [comment]
LGTM

**Pull request merged**